### PR TITLE
Fight and fix multiplayer issues

### PR DESCRIPTION
**Pull request by TSRBerry**
While testing we encounter more and more bugs, so let's create a bugfix PR to fight them properly!

List of issues:

- [x] statusColor defaults to black now instead of yellow
- [x] chest to bot conversion broke
- [x] bot movement is wrong, the bot should move like a farmer
- [x] the usrdisk on the home computer seems to be linked to other players
- [x] hud message shows the wrong name if the player is not online
- [x] Add message when trying to break a bot that belongs to another player
- [x] Sync botName after picking it up
- [x] Sleeping doesn't seem to sync the facingDirection correctly
- [x] Add initially unique bot name
- [x] Allow the bot to be picked up with items again and store the items in the object
- [x] Attempt to add smooth movement for the bot (currently not perfect)
- [x] Other players can't place bots down correctly (they turn invisible)
- [x] After sleeping the location of other players' bots is set incorrectly, which leads to weird issues with the console
- [x] Sync the last inventory slot for hats

### CODE REVIEW

**Comment by TSRBerry** [comment]
Commit 444a973da3490ec98e69694649efd3b4bee522cc should solve the following issue on the [roadmap](https://github.com/JoeStrout/Farmtronics/blob/a6b35f062f1238927add0f1c98e38f43293ffcb6/ROADMAP.md?plain=1#L45).

**Comment by TSRBerry** [comment]
Does movement look okay like this? It's not as smooth for other players as for the owner, but it might be fine? If not I can try to up the sync rate a bit more, but not sure how much better the results will be.

https://user-images.githubusercontent.com/20988865/194386572-665bd8df-4367-4d4e-a28e-f37e6da6b6fe.mp4

**Comment by TSRBerry** [comment]
To break the sleeping bug down:
After sleeping the bots get converted back by the main player. The main player executes the correct constructor and saves the correct location to them. Other players will only execute the first constructor without a `GameLocation`, leaving `position` and `currentLocation` at their default values. The position will be fixed by executing `InitShell()` as soon as the day starts, but the `currentLocation` can't be fixed in the same way, since objects don't store their own location.

So I think there are 2 ways to fix this issue:
1. We could write the `currentLocation` to `modData` and sync it that way.
2. We could assign the `currentLocation` with the `AddBotInstance` message, as it's already required to find the bot on the map  anyways.

**Comment by TSRBerry** [comment]
I just implemented option 2 since it was quick and easy, let me know if you prefer the other option.

File: Farmtronics/Bot/BotFarmer.cs [hunk 1]
```diff
@@ -6,26 +6,24 @@
...
-		
 		// TODO: Figure out if this is needed
 		// 		 The base function calls Game1.player which we need to avoid
 		//		 but for some reason it works just fine at the moment, need to do more testing
 		// public override Vector2 GetToolLocation(bool ignoreClick = false) {
 		// 	Microsoft.Xna.Framework.Rectangle boundingBox = GetBoundingBox();
 		// 	switch (FacingDirection) {
 		// 	case 0:
-		// 		return new Vector2(boundingBox.X + boundingBox.Width / 2, boundingBox.Y - 64);
+		// 		return new Vector2(boundingBox.X + boundingBox.Width / 2, boundingBox.Y - Game1.tileSize);
```
Comment (JoeStrout) [inline Farmtronics/Bot/BotFarmer.cs:16]: Nice improvement.  I never realized this magic number had a name!

File: Farmtronics/Bot/BotObject.cs [hunk 2]
```diff
@@ -163,6 +167,11 @@ class BotObject : StardewValley.Object {
 			if (farmer == null || inventory == null || farmer.CurrentTool == null) return;
 			Vector2 toolLocation = farmer.GetToolLocation(true);
 			ModEntry.instance.Monitor.Log($"UseTool called: {farmer.CurrentTool.Name}[{farmer.CurrentToolIndex}] {toolLocation}");
+			
+			// Check ResourceClamps and current UpgradeLevel before hitting them
```
Comment (JoeStrout) [inline Farmtronics/Bot/BotObject.cs:171]: minor typo: "ResourceClamps" should be "ResourceClumps"

File: Farmtronics/Bot/BotObject.cs [hunk 3]
```diff
@@ -163,6 +167,11 @@ class BotObject : StardewValley.Object {
...
+			var clump = currentLocation.GetCollidingResourceClump(toolLocation);
+			if (clump != null && (clump.GetName().Contains("Stump") || clump.GetName().Contains("Boulder")) && farmer.CurrentTool.UpgradeLevel < 4) return;
```
Comment (JoeStrout) [inline Farmtronics/Bot/BotObject.cs:173]: ...but this is a very nice enhancement. :)

File: Farmtronics/M1/Console.cs [hunk 4]
```diff
@@ -137,12 +137,13 @@ public Console(Shell owner)
 
 		public override void receiveKeyPress(Keys key) {
 			var inp = ModEntry.instance.Helper.Input;
+			// ModEntry.instance.Monitor.Log($"Console.receiveKeyPress({key}) inputMode: {inInputMode}");
```
Comment (JoeStrout) [inline Farmtronics/M1/Console.cs:140]: As there are no changes to this file that aren't either whitespace or now commented out, can you just revert (restore) it?

**Review by JoeStrout — CHANGES_REQUESTED** [review]
Looking great!  Just a couple of very minor changes needed before we merge.

File: Farmtronics/M1/Console.cs (hunk 4 above)
Comment (TSRBerry) [reply Farmtronics/M1/Console.cs]: Didn't even mean to commit them :sweat_smile:

**Comment by JoeStrout** [comment]
Great work again!

**Pull request merged**